### PR TITLE
adding efs creation and efs-on-eks support

### PR DESCRIPTION
**Pull request by dgraeber**
*Issue #, if available:*

*Description of changes:*
- Added a stand-alone module for eks support (in `core`) 
- Added an `integration` module to support EFS on EKS (creation of a storage class)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: modules/core/efs/setup.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+with open("README.md") as fp:
+    long_description = fp.read()
+
+setuptools.setup(
+    version="0.1.0",
+    description="Module creates an EFS service.",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    author="author",
```
Comment (a13zen) [inline modules/core/efs/setup.py:11]: Should we set the author?

File: modules/core/efs/stack_efs.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            vpc_id=self.vpc_id,
+        )
+
+        self.efs_security_group = ec2.SecurityGroup(self, "EFSSecurityGroup", vpc=self.vpc, allow_all_outbound=True)
+        self.efs_filesystem = efs.FileSystem(
+            self,
+            "Filesystem",
+            vpc=self.vpc,
+            security_group=self.efs_security_group,
+            removal_policy=RemovalPolicy.DESTROY,
```
Comment (a13zen) [inline modules/core/efs/stack_efs.py:52]: This we need to pass through as parameter and have the default to retain. Destroy is very dangerous especially if someone isn't aware that it is the default.

File: modules/integration/efs-on-eks/setup.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+with open("README.md") as fp:
+    long_description = fp.read()
+
+setuptools.setup(
+    version="0.1.0",
+    description="A short description of the module.",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    author="author",
```
Comment (a13zen) [inline modules/integration/efs-on-eks/setup.py:11]: Should we add an author?

**Review by a13zen — CHANGES_REQUESTED** [review]
Looks good. Only update requested would be to set default removal policy to `RETAIN` and use a parameter to switch to `DESTROY` where required.

File: modules/core/efs/setup.py (hunk 1 above)
Comment (dgraeber) [reply modules/core/efs/setup.py:11]: Sure

File: modules/integration/efs-on-eks/setup.py (hunk 3 above)
Comment (dgraeber) [reply modules/integration/efs-on-eks/setup.py:11]: sure

File: modules/core/efs/stack_efs.py (hunk 2 above)
Comment (dgraeber) [reply modules/core/efs/stack_efs.py]: Completed

**Review by a13zen — APPROVED** [review]
LGTM 👍

**Pull request merged**